pkg/plugins/policies/meshhttproute/plugin/v1alpha1/gateway.go

```diff
@@ -92,23 +92,23 @@ func generateGatewayRoutes(
 	}
 
 	resources := core_xds.NewResourceSet()
-	routeConfig := plugin_gateway.GenerateRouteConfig(info)
-
 	// Make a pass over the generators for each virtual host.
-	for _, hostInfo := range hostInfos {
-		vh, err := plugin_gateway.GenerateVirtualHost(ctx, info, hostInfo.Host, hostInfo.Entries())
+	for hostname, hostInfos := range hostInfosByHostname {
+		routeConfig := plugin_gateway.GenerateRouteConfig(info.Proxy, info.Listener.Protocol, info.Listener.ResourceName+":"+hostname)
```

Comment:
this is the core change, we create multiple RouteConfigurations

ignore the string concatenations